pkg/operators/wasm/fields.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero"
+	wapi "github.com/tetratelabs/wazero/api"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+)
+
+func (i *wasmOperatorInstance) addFieldFuncs(env wazero.HostModuleBuilder) {
+	env.NewFunctionBuilder().
+		WithGoModuleFunction(
+			wapi.GoModuleFunc(i.fieldAccessorGet),
+			[]wapi.ValueType{
+				wapi.ValueTypeI32, // Accessor
+				wapi.ValueTypeI32, // Data
+				wapi.ValueTypeI32, // Kind
+			},
+			[]wapi.ValueType{wapi.ValueTypeI64}, // Value
+		).
+		Export("fieldAccessorGet")
+
+	env.NewFunctionBuilder().
+		WithGoModuleFunction(
+			wapi.GoModuleFunc(i.fieldAccessorSet),
+			[]wapi.ValueType{
+				wapi.ValueTypeI32, // Accessor
+				wapi.ValueTypeI32, // Data
+				wapi.ValueTypeI32, // Kind
+				wapi.ValueTypeI64, // Value
+			},
+			[]wapi.ValueType{},
+		).
+		Export("fieldAccessorSet")
+}
+
+// fieldAccessorGet returns the field's value.
+// Params:
+// - stack[0]: Field handle
+// - stack[1]: Data handle
+// - stack[2]: Kind
+// Return value:
+// - Uint8 value
```

Comment:
```suggestion
// - Uint64 representation of the value of the field, depending on the type requested, or a pointer
```

Reply:
I changed this function so many times that something had to be wrong for sure.